### PR TITLE
feat(azure): new monitoring check ensuring storage account with logs private

### PR DESCRIPTION
**Pull request by Hugo966**
### Context

New monitor check 5.1.3

### Description

This check ensures that you have "Allow Blob anonymous access" disabled in the storage accounts in which you have logs stored.

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: prowler/providers/azure/services/monitor/monitor_ensure_storage_container_with_activity_log_private/monitor_ensure_storage_container_with_activity_log_private.metadata.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{
+  "Provider": "azure",
+  "CheckID": "monitor_ensure_storage_container_with_activity_log_private",
+  "CheckTitle": "Ensure Diagnostic Setting captures appropriate categories",
+  "CheckType": [],
+  "ServiceName": "monitor",
+  "SubServiceName": "Configuring Diagnostic Settings",
```
Comment (sergargar) [inline prowler/providers/azure/services/monitor/monitor_ensure_storage_container_with_activity_log_private/monitor_ensure_storage_container_with_activity_log_private.metadata.json:7]:
```suggestion
```

File: prowler/providers/azure/services/monitor/monitor_ensure_storage_container_with_activity_log_private/monitor_ensure_storage_container_with_activity_log_private.metadata.json [hunk 2]
```diff
@@ -0,0 +1,30 @@
+{
+  "Provider": "azure",
+  "CheckID": "monitor_ensure_storage_container_with_activity_log_private",
+  "CheckTitle": "Ensure Diagnostic Setting captures appropriate categories",
```
Comment (MrCloudSec) [inline prowler/providers/azure/services/monitor/monitor_ensure_storage_container_with_activity_log_private/monitor_ensure_storage_container_with_activity_log_private.metadata.json]: This is not for this check

File: prowler/providers/azure/services/monitor/monitor_ensure_storage_container_with_activity_log_private/monitor_ensure_storage_container_with_activity_log_private.metadata.json [hunk 3]
```diff
@@ -0,0 +1,30 @@
+{
+  "Provider": "azure",
+  "CheckID": "monitor_ensure_storage_container_with_activity_log_private",
```
Comment (MrCloudSec) [inline prowler/providers/azure/services/monitor/monitor_ensure_storage_container_with_activity_log_private/monitor_ensure_storage_container_with_activity_log_private.metadata.json]:
```suggestion
  "CheckID": "monitor_storage_container_with_activity_logs_is_private",
```

File: prowler/providers/azure/services/monitor/monitor_ensure_storage_container_with_activity_log_private/monitor_ensure_storage_container_with_activity_log_private.metadata.json [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  "ResourceType": "Monitor",
+  "Description": "The storage account container containing the activity log export should not be publicly accessible.",
+  "Risk": "Allowing public access to activity log content may aid an adversary in identifying weaknesses in the affected account's use or configuration.",
+  "RelatedUrl": "https://learn.microsoft.com/en-us/azure/azure-monitor/essentials/diagnostic-settings",
+  "Remediation": {
+    "Code": {
+      "CLI": "az storage container set-permission --name insights-activity-logs --account-name <Storage Account Name> --sas-token <SAS token> --public-access off",
+      "NativeIaC": "",
+      "Other": "https://www.trendmicro.com/cloudoneconformity-staging/knowledge-base/azure/Monitor/check-for-publicly-accessible-activity-log-storage-container.html",
+      "Terraform": "https://docs.bridgecrew.io/docs/ensure-the-storage-container-storing-the-activity-logs-is-not-publicly-accessible"
```
Comment (MrCloudSec) [inline prowler/providers/azure/services/monitor/monitor_ensure_storage_container_with_activity_log_private/monitor_ensure_storage_container_with_activity_log_private.metadata.json]:
```suggestion
      "Terraform": "https://docs.bridgecrew.io/docs/ensure-the-storage-container-storing-the-activity-logs-is-not-publicly-accessible#terraform"
```

File: prowler/providers/azure/services/monitor/monitor_ensure_storage_container_with_activity_log_private/monitor_ensure_storage_container_with_activity_log_private.metadata.json [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+  "SubServiceName": "Configuring Diagnostic Settings",
+  "ResourceIdTemplate": "",
+  "Severity": "high",
+  "ResourceType": "Monitor",
+  "Description": "The storage account container containing the activity log export should not be publicly accessible.",
+  "Risk": "Allowing public access to activity log content may aid an adversary in identifying weaknesses in the affected account's use or configuration.",
+  "RelatedUrl": "https://learn.microsoft.com/en-us/azure/azure-monitor/essentials/diagnostic-settings",
+  "Remediation": {
+    "Code": {
+      "CLI": "az storage container set-permission --name insights-activity-logs --account-name <Storage Account Name> --sas-token <SAS token> --public-access off",
```
Comment (MrCloudSec) [inline prowler/providers/azure/services/monitor/monitor_ensure_storage_container_with_activity_log_private/monitor_ensure_storage_container_with_activity_log_private.metadata.json]:
```suggestion
      "CLI": "az storage container set-permission --name insights-activity-logs --account-name <Storage Account Name> --public-access off",
```

File: prowler/providers/azure/services/monitor/monitor_ensure_storage_container_with_activity_log_private/monitor_ensure_storage_container_with_activity_log_private.metadata.json (hunk 1 above)
Comment (MrCloudSec) [inline prowler/providers/azure/services/monitor/monitor_ensure_storage_container_with_activity_log_private/monitor_ensure_storage_container_with_activity_log_private.metadata.json]:
```suggestion
  "SubServiceName": "",
```

File: prowler/providers/azure/services/monitor/monitor_ensure_storage_container_with_activity_log_private/monitor_ensure_storage_container_with_activity_log_private.metadata.json [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+      "NativeIaC": "",
+      "Other": "https://www.trendmicro.com/cloudoneconformity-staging/knowledge-base/azure/Monitor/check-for-publicly-accessible-activity-log-storage-container.html",
+      "Terraform": "https://docs.bridgecrew.io/docs/ensure-the-storage-container-storing-the-activity-logs-is-not-publicly-accessible"
+    },
+    "Recommendation": {
+      "Text": "1. From Azure Home select the Portal Menu 2. Search for Storage Accounts to access Storage account blade 3. Click on the storage account name 4. Click on Configuration under settings 5. Select Enabled under 'Allow Blob public access'",
+      "Url": "https://learn.microsoft.com/en-us/python/api/azure-mgmt-monitor/azure.mgmt.monitor.v2021_05_01_preview.operations.diagnosticsettingsoperations?view=azure-python"
```
Comment (MrCloudSec) [inline prowler/providers/azure/services/monitor/monitor_ensure_storage_container_with_activity_log_private/monitor_ensure_storage_container_with_activity_log_private.metadata.json]: Is there any other URL?

File: prowler/providers/azure/services/monitor/monitor_storage_container_with_activity_logs_is_private/monitor_storage_container_with_activity_logs_is_private.py [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+            report.status = "PASS"
+            report.subscription = subscription_name
+            report.resource_name = "Monitor"
+            report.resource_id = "Monitor"
+            report.status_extended = f"Blob public access disabled in storage account storing activity log in subscription {subscription_name} or not necessary."
+            for diagnostic_setting in diagnostic_settings:
+                sa_id = diagnostic_setting.storage_account_id  # sa = storage account
+                elements = sa_id.split("/")
+                sa_name = elements[8]
+                for sub, storage_accounts in storage_client.storage_accounts.items():
```
Comment (MrCloudSec) [inline prowler/providers/azure/services/monitor/monitor_storage_container_with_activity_logs_is_private/monitor_storage_container_with_activity_logs_is_private.py]:
```suggestion
                for storage_account in storage_client.storage_accounts[subscription_name]:
```

File: prowler/providers/azure/services/monitor/monitor_storage_container_with_activity_logs_is_private/monitor_storage_container_with_activity_logs_is_private.py [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+                    for storage_account in storage_accounts:
```
Comment (MrCloudSec) [inline prowler/providers/azure/services/monitor/monitor_storage_container_with_activity_logs_is_private/monitor_storage_container_with_activity_logs_is_private.py]:
```suggestion
```

File: prowler/providers/azure/services/monitor/monitor_storage_container_with_activity_logs_is_private/monitor_storage_container_with_activity_logs_is_private.py [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+                        if storage_account.name == sa_name:
+                            if storage_account.allow_blob_public_access:
+                                report.status = "FAIL"
+                                report.status_extended = f"Blob public access enabled in storage account storing activity log in subscription {subscription_name}."
```
Comment (MrCloudSec) [inline prowler/providers/azure/services/monitor/monitor_storage_container_with_activity_logs_is_private/monitor_storage_container_with_activity_logs_is_private.py]:
```suggestion
                                report.status_extended = f"Blob public access enabled in storage account {storage_account.name} storing activity logs in subscription {subscription_name}."
```

File: prowler/providers/azure/services/monitor/monitor_storage_container_with_activity_logs_is_private/monitor_storage_container_with_activity_logs_is_private.py [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+        for (
+            subscription_name,
+            diagnostic_settings,
+        ) in monitor_client.diagnostics_settings.items():
+            report = Check_Report_Azure(self.metadata())
+            report.status = "PASS"
+            report.subscription = subscription_name
+            report.resource_name = "Monitor"
+            report.resource_id = "Monitor"
+            report.status_extended = f"Blob public access disabled in storage account storing activity log in subscription {subscription_name} or not necessary."
```
Comment (MrCloudSec) [inline prowler/providers/azure/services/monitor/monitor_storage_container_with_activity_logs_is_private/monitor_storage_container_with_activity_logs_is_private.py]: Please, create a finding per storage account storing activity logs.

File: prowler/providers/azure/services/monitor/monitor_storage_container_with_activity_logs_is_private/monitor_storage_container_with_activity_logs_is_private.py [hunk 11]
```diff
@@ -0,0 +1,38 @@
+from prowler.lib.check.models import Check, Check_Report_Azure
+from prowler.providers.azure.services.monitor.monitor_client import monitor_client
+from prowler.providers.azure.services.storage.storage_client import storage_client
+
+
+class monitor_storage_container_with_activity_logs_is_private(Check):
```
Comment (MrCloudSec) [inline prowler/providers/azure/services/monitor/monitor_storage_container_with_activity_logs_is_private/monitor_storage_container_with_activity_logs_is_private.py]:
```suggestion
class monitor_storage_account_with_activity_logs_is_private(Check):
```

File: prowler/providers/azure/services/monitor/monitor_storage_account_with_activity_logs_is_private/monitor_storage_account_with_activity_logs_is_private.py [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+        ) in monitor_client.diagnostics_settings.items():
+            if not diagnostic_settings:
+                report = Check_Report_Azure(self.metadata())
+                report.subscription = subscription_name
+                report.resource_name = "Monitor"
+                report.resource_id = "Monitor"
+                report.status = "PASS"
+                report.status_extended = f"Blob public access disabled in storage account storing activity log in subscription {subscription_name} or not necessary."
+                findings.append(report)
+                continue
```
Comment (MrCloudSec) [inline prowler/providers/azure/services/monitor/monitor_storage_account_with_activity_logs_is_private/monitor_storage_account_with_activity_logs_is_private.py]: Is this needed?

File: prowler/providers/azure/services/monitor/monitor_storage_account_with_activity_logs_is_private/monitor_storage_account_with_activity_logs_is_private.py [hunk 13]
```diff
@@ -0,0 +1,44 @@
...
+                        report.resource_name = "Monitor"
+                        report.resource_id = "Monitor"
+                        if storage_account.allow_blob_public_access:
+                            report.status = "FAIL"
+                            report.status_extended = f"Blob public access enabled in storage account {storage_account.name} storing activity logs in subscription {subscription_name}."
+                            findings.append(report)
+                        else:
+                            report.status = "PASS"
+                            report.status_extended = f"Blob public access disabled in storage account {storage_account.name} storing activity logs in subscription {subscription_name}."
+                            findings.append(report)
```
Comment (MrCloudSec) [inline prowler/providers/azure/services/monitor/monitor_storage_account_with_activity_logs_is_private/monitor_storage_account_with_activity_logs_is_private.py]: Use only one append for the if and else

File: prowler/providers/azure/services/monitor/monitor_storage_account_with_activity_logs_is_private/monitor_storage_account_with_activity_logs_is_private.py (hunk 12 above)
Comment (Hugo966) [reply prowler/providers/azure/services/monitor/monitor_storage_account_with_activity_logs_is_private/monitor_storage_account_with_activity_logs_is_private.py]: When there are not diagnostic settings, the check must pass, this is for those cases, I did a test with no diagnostic settings and I noticed.

Comment (MrCloudSec) [reply prowler/providers/azure/services/monitor/monitor_storage_account_with_activity_logs_is_private/monitor_storage_account_with_activity_logs_is_private.py]: I would say that this is not a PASS nor FAIL, there is no storage accounts where a finding can be applied.

Comment (Hugo966) [reply prowler/providers/azure/services/monitor/monitor_storage_account_with_activity_logs_is_private/monitor_storage_account_with_activity_logs_is_private.py]: Okey

**Review by MrCloudSec — DISMISSED** [review]
Good job!

**Review by MrCloudSec — DISMISSED** [review]

**Review by MrCloudSec — APPROVED** [review]

**Pull request merged**